squidDrone/src/propulsion/little_bee_20_a.cpp

```diff
@@ -0,0 +1,60 @@
+#include "little_bee_20_a.hpp"
+#include "globals.hpp"
+
+namespace propulsion{
+
+  const auto LittleBee20A::SetPulseDuration(int pulse_duration, int repetition_period) noexcept -> types::HalError{
+    types::HalError error_state;
+    const bool pulse_limit_breach = pulse_duration > ONESHOT_125_MAX_PULSE_DURATION_IN_US_ || pulse_duration < ONESHOT_125_MIN_PULSE_DURATION_IN_US_;
+    const bool period_limit_breach = repetition_period < pulse_duration || repetition_period > MAX_REPETITION_PERIOD_;
+    if(pulse_limit_breach || period_limit_breach){
+      return types::HalError::PARAMETER_ERROR;
+    }else{
+      if(!timer_is_configured_){
+        auto configuration_error_state = ConfigureTimer();
+        if(configuration_error_state != types::HalError::WORKING){
+          return types::HalError::CONFIG_ERROR;
+        }
+      }
+      const std::uint32_t pulse_duration_in_ticks = static_cast<std::uint32_t>(pulse_duration * TICKS_PER_MICROSECOND_);
+      const std::uint32_t period = static_cast<std::uint32_t>(repetition_period * TICKS_PER_MICROSECOND_);
+      error_state = SetPwm(period, pulse_duration_in_ticks);
+    }
+    return error_state;
+  }
+
+  const auto LittleBee20A::ConfigureTimer() noexcept -> types::HalError{
+    if(HAL_TIM_PWM_Stop(timer_, channel_) != HAL_OK){
+      return types::HalError::CONFIG_ERROR;
+    }
+    uint32_t prescaler_10_mhz = __HAL_TIM_CALC_PSC(TIMER_CLOCK, TARGET_TIMER_CLOCK_RATE_); //calculate timer input clock
```

Comment:
missing std:: before uint32_t

Reply:
fixed in [this commit](https://github.com/elheck/Squiddrone/pull/168/commits/04989db9ed3460ed454f9233b555d70d0787455c)